listenbrainz/webserver/static/js/src/types.d.ts

```diff
@@ -13,38 +48,75 @@ declare type ListenBrainzUser = {
 declare type ListenType = "single" | "playingNow" | "import";
 
 declare type SpotifyPlayDirection = "up" | "down" | "hidden";
-// TODO: make this type specific
-declare type SubmitListensPayload = any;
+
+declare type SubmitListensPayload = {
+  listen_type: "single" | "playing_now" | "import";
+  payload: Array<Listen>;
+};
 
 declare type SpotifyUser = {
   access_token: string;
   permission: string;
 };
 
-declare type SpotifyPermission = any; // TODO
-declare type SpotifyTrack = any; // TODO
-declare type SpotifyArtist = any; // TODO
-declare type SpotifyDeviceID = any; // TODO
-declare interface SpotifyImage {
+declare type SpotifyPermission = string;
```

Comment:
we only have a few specific permissions for spotify. 

Let's make the type reflect that. (https://github.com/metabrainz/listenbrainz-server/blob/5b1a66ea/listenbrainz/domain/spotify.py#L14)